### ISSUE TITLE
Extreme input lag

### ISSUE THREAD

**Issue by aregularplayer** [haxball/haxball-issues#1101]
Hey, i want to talk about this extreme issue that i have when playing haxball.
Since 2 months ago i started to have some extreme input lag that is basically my "player" is very heavy and slow.
Everyday this is worst, i consistently have to increase the extrapolation and its very annoying, right now i use 120 extrapolation and if it continues like this, i'll have to use 200 extrapolation and since it still increases , the game will be unplayable and i will leave this game that i play since a long time ago and i like.

I've tested on my brother's pc and its the same issue.
I've tested with my pc on another house and its the same.
Meanwhile i've put a ssd and made a fresh install of windows 10 on my pc and its the same.

What do you think it is? And what should i do?

**Comment by aregularplayer** [comment]
Edit: basro you should really talk about this.
Its a issue that many players have and dont know how to resolve

**Comment by basro** [comment]
I am not able to reproduce this.
I've tested on Firefox and Chrome, in two different computers (one of them being a fairly old laptop).

There haven't been significant changes to haxball in the last 2 months. If this is happening it must be caused by a change in the OS or the browser. Unless I'm able to reproduce it there's very low chances that I could do anything to fix it.

Also, are you able to 100% discard the possibility that this is a change in your perception and not in the game? If you get used to playing with extrapolation it will make the gameplay feel very "heavy" when you disable it. Does kicking the ball also feel laggy or is it just the player movement?

**Comment by aregularplayer** [comment]
Hey thanks for the reply, i've tested on multiple browsers and the result its the same.
Yes im 99.9 % sure this isn't a change of perception , also i can't get used to the sensitivity (for example one day i play with 80 extrapolation and the next day that 80 extrapolation will feel heavy so i always have to change the extrapolation even in middle of games ).
And both the player kick and movement are delayed and also something i forgot to say is the sound when the player kicks the ball is delayed/late.

**Comment by basro** [comment]
Sounds are not predicted so at least that part is expected/normal.

**Comment by aregularplayer** [comment]
But what are you going to do?
Well if this input lag is day after day worst, probably the day will be unplayable, even with the max possible extrapolation.

**Comment by rantz4** [comment]
wellll, it's already a issue that most of the communities/players complain about. It is very annoying to be forced to play with extrapolation tbh. When we players have extrapolation, the game becomes something else fr, everything is shaking and randomly played and also feeling is still bad. But on the other side when you don't have extrapolation, game seems nice enough but you can't control your player, your movements/kicks all are delayed. So idk what actually can be done but something should be done the game is not stable, a lot of players from flash already retired just cus of this problem and there will be more retiring.  If you at least try to fix this issue, we all would be helping you at any time.

**Comment by aregularplayer** [comment]
mate , my case its like this for example one day im playing with 80 extrapolation , the gameplay feels smooth and stuff and the next day it doesnt feel smooth and its more delayed , so everyday i have to put more extrapolation so i can't have a stable gameplay and i can't get used to this.
Before this the game was so much fun and i played almost everyday but now i only play leagues and i can't have fun because i can't play my best

**Comment by 9999OldSchool** [comment]
I agree. I've been playing the game these days. I don't feel consistent movement at all even though I use extrapolation. It's just different feelings every day, and it's really frustrating. I also have Win 10, but doesn't really help at all. 
Also, sometimes I would practice a bit before an official match (like fs), and then join the room before an official match and it's just completely different feelings in 1 day. It wasn't even smooth, I couldn't move myself like holy shit. 

> this is a video also how the game is like nowadays; https://www.youtube.com/watch?v=bKi71IruuIA&feature=emb_title

**Comment by aregularplayer** [comment]
Observation:
I think you can reproduce this basro. 
At least for me the game feels like it runs at negative extrapolation (for example when im with 0 extrapolation it feels like -100  extrapolation)

**Comment by reyeso23** [comment]
Did you fix it?
